custom_components/frigate/views.py

```diff
@@ -224,7 +225,7 @@ def _permit_request(self, request: web.Request, config_entry: ConfigEntry) -> bo
         return bool(config_entry.options.get(CONF_NOTIFICATION_PROXY_ENABLE, True))
 
 
-def _init_header(request: web.Request) -> CIMultiDict | dict[str, str]:
+def _init_header(request: web.Request, upstream_host: str) -> CIMultiDict | dict[str, str]:
```

Comment:
The `host` method on a URL can return empty, so this will fail type checks. You'd need update upstream_host to be `str | None`, and then an if statement below to not populate it if it's `None`.